app/scripts/models/reliers/oauth.js

```diff
@@ -171,6 +174,14 @@ define([
           }
           throw err;
         });
+    },
+
+    isTrusted: function () {
+      return this.get('trusted');
+    },
+
+    needsPermissions: function (account) {
+      return ! this.isTrusted() && ! account.hasGrantedPermissions(this.get('clientId'), this.get('scope'));
```

Comment:
Minor nit, but the intention could probably be easier to understand if it were re-written a bit like:

```js
if (this.isTrusted()) {
  return false;
}
return ! account.hasGrantedPermissions(...
```

My guess is uglify will combine the two statements into something similar to what you have here.